pkg/cmd/adm/register_member.go

```diff
@@ -120,66 +114,198 @@ func registerMemberCluster(ctx *extendedCommandContext, newCommand client.Comman
+		ctx.Printlnf("The %s ServiceAccount is not present in the %s cluster.", toolchainClusterSAKey, sourceClusterType)
+		ctx.Printlnf("Please check the %[1]s ToolchainCluster ServiceAccount in the %[2]s %[3]s cluster or the deployment of the %[3]s operator.", toolchainClusterSAKey, sourceClusterDetails.apiEndpoint, sourceClusterType)
+		return err
 	}
+	// source cluster details
+	ctx.Printlnf("The source cluster name: %s", sourceClusterDetails.toolchainClusterName)
+	ctx.Printlnf("The API endpoint of the source cluster: %s", sourceClusterDetails.apiEndpoint)
+
+	// target to details
+	ctx.Printlnf("The name of the target cluster: %s", targetClusterDetails.toolchainClusterName)
+	ctx.Printlnf("The API endpoint of the target cluster: %s", targetClusterDetails.apiEndpoint)
 
-	script, err := downloadScript(term)
+	// generate a token that will be used for the kubeconfig
+	sourceTargetRestClient, err := newRestClient(sourceClusterDetails.kubeConfig)
 	if err != nil {
 		return err
 	}
-	args := []string{script.Name(), "--type", joiningClusterType.String(), "--host-kubeconfig", hostKubeconfig, "--host-ns", hostNs, "--member-kubeconfig", memberKubeconfig, "--member-ns", memberNs}
-	if len(nameSuffix) > 0 {
-		args = append(args, "--multi-member", nameSuffix)
+	token, err := generate.GetServiceAccountToken(sourceTargetRestClient, toolchainClusterSAKey, TokenExpirationDays)
+	if err != nil {
+		return err
 	}
-	if useLetsEncrypt {
-		args = append(args, "--lets-encrypt")
+	// TODO drop this part together with the --lets-encrypt flag and start loading certificate from the kubeconfig as soon as ToolchainCluster controller supports loading certificates from kubeconfig
+	var insecureSkipTLSVerify bool
+	if v.args.useLetsEncrypt {
+		ctx.Printlnf("using let's encrypt certificate")
+		insecureSkipTLSVerify = false
+	} else {
+		ctx.Printlnf("setting insecure skip tls verification flags")
+		insecureSkipTLSVerify = true
+	}
+	// generate the kubeconfig that can be used by target cluster to interact with the source cluster
+	generatedKubeConfig := generateKubeConfig(token, sourceClusterDetails.apiEndpoint, sourceClusterDetails.namespace, insecureSkipTLSVerify)
+	generatedKubeConfigFormatted, err := clientcmd.Write(*generatedKubeConfig)
+	if err != nil {
+		return err
 	}
-	term.Printlnf("Command to be called: bash %s\n", strings.Join(args, " "))
-	bash := newCommand("bash", args...)
-	bash.Stdout = os.Stdout
-	bash.Stderr = os.Stderr
-	return bash.Run()
-}
 
-func downloadScript(term ioutils.Terminal) (*os.File, error) {
-	resp, err := http.Get(AddClusterScriptURL)
+	// Create or Update the secret on the targetCluster
+	secretName := toolchainClusterSAKey.Name + "-" + sourceClusterDetails.toolchainClusterName
+	ctx.Printlnf("creating secret %s/%s in the %s cluster", targetClusterDetails.namespace, secretName, targetClusterDetails)
```

Comment:
this should be
```suggestion
	ctx.Printlnf("creating secret %s/%s in the %s cluster", targetClusterDetails.namespace, secretName, sourceClusterType.TheOtherType())
```
otherwise it prints the whole targetClusterDetails object

Reply:
thanks!